freeipa/src/main/java/com/sequenceiq/freeipa/sync/ProviderChecker.java

```diff
@@ -91,8 +101,14 @@ private InstanceStatus updateStatuses(CloudVmInstanceStatus vmInstanceStatus, In
             default:
                 LOGGER.info(":::Auto sync::: the '{}' status is not converted", vmInstanceStatus.getStatus());
         }
-        if (updateStatus) {
-            instanceMetaDataService.save(instanceMetaData);
+        if (!flowLogService.isOtherFlowRunning(stack.getId())) {
```

Comment:
if the flow is finished while you iterate over the list, what will happen?

Reply:
true!